### DISCUSSION TITLE
how to use url and put images and fonts into folder

### DISCUSSION THREAD

**Discussion by doctorzan** [Appsilon/rhino#568] [Q&A]
I use this code to import js
```
shiny::includeScript(file.path(getwd(), "app/static/js/main.js"))
```
in this situtaion I can't use url code in js
```
url('images/apc.jpg')
```
Questions:

In Rhino, where should I place my images folder to use URLs in my JavaScript code correctly? Is there any issue with this code snippet?

**Comment by kamilzyla** [comment]
Hi @doctorzan and thanks for using Rhino!

The preferred way of using JavaScript in Rhino is to keep it in `app/js/index.js`. After editing this file you need to run `rhino::build_js()` to create a minified `app/static/js/app.min.js` file, which is automatically loaded by Rhino. Check out our [How-to: Write JavaScript code](https://appsilon.github.io/rhino/articles/how-to/write-javascript-code.html).

Keep your images in `app/static` and use relative paths with `/static/` prefix to refer to them. This should work whether you want to use it in JS or R. For example, to display `app/static/images/apc.jpg` in your app, use:
```r
img(src = "/static/images/apc.jpg")
```

If this is not what you need, please tell me a bit more about what you're trying to achieve. Is there a reason you want to use `shiny::includeScript()` instead of the method described above? What is the `url()` function in your snippet?

**Reply by doctorzan** [reply]
Thank you
